pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -1272,3 +1272,11 @@ func setTerminationReasonByPhase(phase v1alpha1.MachinePhase, terminationConditi
 		terminationCondition.Message = "Machine Controller is scaling down machine"
 	}
 }
+
+func getProviderID(machine *v1alpha1.Machine) string {
+	return machine.Spec.ProviderID
```

Comment:
What happens if any of the values in the flow isn't set? I hope it doesn't panic.